Source/WTF/wtf/text/StringCommon.h

```diff
@@ -1163,10 +1163,102 @@ inline std::span<const char> span(const char* string)
     return { string, string ? strlen(string) : 0 };
 }
 
+#if CPU(ARM64)
+
+ALWAYS_INLINE uint8x16_t loadBulk(const uint8_t* ptr)
+{
+    return vld1q_u8(ptr);
+}
+
+ALWAYS_INLINE uint16x8_t loadBulk(const uint16_t* ptr)
+{
+    return vld1q_u16(ptr);
+}
+
+ALWAYS_INLINE uint8x16_t mergeBulk(uint8x16_t accumulated, uint8x16_t input)
+{
+    return vorrq_u8(accumulated, input);
+}
+
+ALWAYS_INLINE uint16x8_t mergeBulk(uint16x8_t accumulated, uint16x8_t input)
+{
+    return vorrq_u16(accumulated, input);
+}
+
+ALWAYS_INLINE bool isNonZeroBulk(uint8x16_t accumulated)
+{
+    return vmaxvq_u8(accumulated);
+}
+
+ALWAYS_INLINE bool isNonZeroBulk(uint16x8_t accumulated)
+{
+    return vmaxvq_u16(accumulated);
+}
+
+template<LChar character, LChar... characters>
+ALWAYS_INLINE uint8x16_t compareBulk(uint8x16_t input)
+{
+    auto result = vceqq_u8(input, vmovq_n_u8(character));
+    if constexpr (!sizeof...(characters))
+        return result;
+    else
+        return mergeBulk(result, compareBulk<characters...>(input));
+}
+
+template<UChar character, UChar... characters>
+ALWAYS_INLINE uint16x8_t compareBulk(uint16x8_t input)
+{
+    auto result = vceqq_u16(input, vmovq_n_u16(character));
+    if constexpr (!sizeof...(characters))
+        return result;
+    else
+        return mergeBulk(result, compareBulk<characters...>(input));
```

Comment:
I see. For each char (8 or 16), extend it to 128 with copies and then compare the input bulk (128). Then, once found a match we recorded and merge the (0xFF) in the result. Finally, if the result is not all 0's then it indicates at least one match.

Reply:
Right